lib/msf/core/post/file.rb

```diff
@@ -578,6 +579,32 @@ def copy_file(src_file, dst_file)
 
 protected
 
+  def _read_file_powershell(filename)
+    data = ''
+    offset = 0
+    chunk_size = 1024
+    loop do
+      chunk = _read_file_powershell_fragment(filename, chunk_size, offset)
+      break if chunk.nil?
+      data << chunk
+      offset += chunk_size
+      break if chunk.length < chunk_size
+    end
+    return data
+  end
+
+  def _read_file_powershell_fragment(filename, chunk_size, offset=0)
+    b64_data= cmd_exec("$mstream = [System.IO.MemoryStream]::new();\
+      $gzipstream = [System.IO.Compression.GZipStream]::new($mstream, [System.IO.Compression.CompressionMode]::Compress);\
+      $get_bytes = [System.IO.File]::ReadAllBytes(\"#{filename}\")[#{offset}..#{offset + chunk_size -1}];\
+      $gzipstream.Write($get_bytes, 0 , $get_bytes.Length);\
+      $gzipstream.Close();\
+      [Convert]::ToBase64String($mstream.ToArray())")
+    return nil if b64_data.empty?
```

Comment:
I added this check because I noticed when the file size was a multiple of the chunk_size the last chunk was an empty string which will raise an error in the gunzip process.